### PR TITLE
Add working example of consuming as bzlmod

### PR DESCRIPTION
**Pull request by JohnRudolfLewis**
- Add an example for how to consume via bzlmod
- Fixes https://github.com/mvukov/rules_ros/issues/32 by removing the dev_dependency = True from rules_python
- bumps rules_python to the latest version
- makes module name and repo_name consistent so local and remote builds work properly

Continues the work started by @hofbi in GH-33

### CODE REVIEW

**Comment by JohnRudolfLewis** [comment]
@hofbi I was able to reproduce the inability to build with your changes in GH-33. A few more changes were needed.

File: MODULE.bazel [hunk 1]
```diff
@@ -42,9 +41,9 @@ use_repo(
 bazel_dep(name = "rules_boost")
 archive_override(
     module_name = "rules_boost",
-    integrity = "sha256-I1iTdF3qckTmDRab+0yjA37Iya9AOGvEGJVoPtfpADM=",
-    strip_prefix = "rules_boost-42d8155d8f20a1aee8ee20b7903a495bdfb9befd",
-    urls = "https://github.com/nelhage/rules_boost/archive/42d8155d8f20a1aee8ee20b7903a495bdfb9befd.zip",
+     integrity = "sha256-EEmldYvkzka7KLDMgziSgCXDWK9J04/F3DBdFnv+188=",
```
Comment (hofbi) [inline MODULE.bazel:44]:
```suggestion
    integrity = "sha256-EEmldYvkzka7KLDMgziSgCXDWK9J04/F3DBdFnv+188=",
```

File: examples/bzlmod/MODULE.bazel [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+#    commit = "160940582fea35a2335d04183e0c540c1209d27e", # TODO update 
+#)
+
+bazel_dep(name = "rules_boost")
+archive_override(
+    module_name = "rules_boost",
+    integrity = "sha256-EEmldYvkzka7KLDMgziSgCXDWK9J04/F3DBdFnv+188=",
+    strip_prefix = "rules_boost-8fa193c4e21daaa2d46ff6b9c2b5a2de70b6caa1",
+    urls = "https://github.com/nelhage/rules_boost/archive/8fa193c4e21daaa2d46ff6b9c2b5a2de70b6caa1.zip",
+)
```
Comment (hofbi) [inline examples/bzlmod/MODULE.bazel:25]:
```suggestion
)

```

File: examples/bzlmod/MODULE.bazel [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+local_path_override(
+    module_name = "com_github_mvukov_rules_ros",
+    path = "../..",
+)
+# but normal usage would be to use a git override
+#git_override(
+#    module_name = "com_github_mvukov_rules_ros",
+#    remote = "https://github.com/mvukov/rules_ros.git",
+#    commit = "160940582fea35a2335d04183e0c540c1209d27e", # TODO update 
+#)
```
Comment (hofbi) [inline examples/bzlmod/MODULE.bazel:17]: My recommendation would be to use https://bazel.build/rules/lib/globals/module#archive_override with https://github.com/mvukov/rules_ros//archive/SHA_OF_COMMIT.zip as url.

The zip file to only download what you need and `archive_override` as you want to consume it via bzlmod

**Review by hofbi — COMMENTED** [review]
Thanks for providing the fix. I added it to https://github.com/mvukov/rules_ros/pull/33, let's see who get's in first.

File: examples/bzlmod/MODULE.bazel [hunk 4]
```diff
@@ -9,12 +9,13 @@ local_path_override(
...
-#    module_name = "com_github_mvukov_rules_ros",
-#    remote = "https://github.com/mvukov/rules_ros.git",
-#    commit = "160940582fea35a2335d04183e0c540c1209d27e", # TODO update 
-#)
+# but normal usage would be to use an archive override
+# archive_override(
+#     module_name = "com_github_mvukov_rules_ros",
+#     integrity = "sha256-zIzL/BydBIPyok6HhmaKbKIA4SL6qE9buk2Sz+E8LeQ=", # TODO Update
+#     strip_prefix = "rules_ros-8223b6f7dc6f35cc7aca74cb9a09394a76a149bb", # TODO Update
+#     urls = "https://github.com/mvukov/rules_ros//archive/8223b6f7dc6f35cc7aca74cb9a09394a76a149bb.zip" # TODO Update
```
Comment (hofbi) [inline examples/bzlmod/MODULE.bazel:17]:
```suggestion
#     urls = "https://github.com/mvukov/rules_ros/archive/8223b6f7dc6f35cc7aca74cb9a09394a76a149bb.zip" # TODO Update
```

**Comment by mvukov** [comment]
Addressed in #43 , I believe.

**Pull request closed without merging**